cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -212,6 +239,15 @@ protected AbstractContext(Program program, ProgramOptions programOptions, CConfi
     }
   }
 
+  @Nullable
+  public static TriggeringScheduleInfo getTriggeringScheduleInfo(ProgramOptions programOptions) {
+    String scheduleInfoString = programOptions.getArguments().getOption(ProgramOptionConstants.TRIGGERING_SCHEDULE_INFO);
+    if (scheduleInfoString == null) {
+      return null;
+    }
+    return GSON.fromJson(scheduleInfoString, TriggeringScheduleInfo.class);
```

Comment:
Wait..I see it gets called in constructor, but why this is a public static method?? should be a private static one.